src/libraries/System.Private.CoreLib/src/System/Reflection/Emit/ILGenerator.cs

```diff
@@ -200,6 +201,61 @@ public virtual LocalBuilder DeclareLocal(Type localType)
+        /// <param name="startLine">The line where the sequence point begins.</param>
+        /// <param name="startColumn">The column in the line where the sequence point begins.</param>
+        /// <param name="endLine">The line where the sequence point ends.</param>
+        /// <param name="endColumn">The column in the line where the sequence point ends.</param>
+        /// <exception cref="ArgumentNullException"><paramref name="document"/> is <see langword="null"/>.</exception>
+        /// <exception cref="ArgumentException"><paramref name="document"/> is not valid.</exception>
+        /// <exception cref="ArgumentOutOfRangeException">
+        /// <paramref name="startLine"/> is not within range [0, 0x20000000).
+        /// <paramref name="endLine"/> is not within range [0, 0x20000000) or lower than <paramref name="startLine"/>.
+        /// <paramref name="startColumn"/> is not within range [0, 0x10000).
+        /// <paramref name="endLine"/> is not within range [0, 0x10000) or
+        /// <paramref name="startLine"/> equal to <paramref name="endLine"/> and <paramref name="endLine"/> lower than or equal to <paramref name="startLine"/>.
+        /// </exception>
+        public void MarkSequencePoint(ISymbolDocumentWriter document, int startLine, int startColumn, int endLine, int endColumn)
+        {
+            ArgumentNullException.ThrowIfNull(document);
+
+            if (startLine < 0 || startLine >= 0x20000000)
+            {
+                throw new ArgumentOutOfRangeException(nameof(startLine));
+            }
+
+            if (endLine < 0 || endLine > 0x20000000 || startLine > endLine)
+            {
+                throw new ArgumentOutOfRangeException(nameof(endLine));
+            }
+
+            if (startColumn < 0 || startColumn > 0x10000)
+            {
+                throw new ArgumentOutOfRangeException(nameof(startColumn));
+            }
+
+            if (endColumn < 0 || endColumn > 0x10000 ||
+                (startLine == endLine && startLine != 0xfeefee && startColumn >= endColumn))
+            {
+                throw new ArgumentOutOfRangeException(nameof(endColumn));
+            }
+
+            MarkSequencePointCore(document, startLine, startColumn, endLine, endColumn);
+        }
+
+        /// <summary>
+        /// When overridden in a derived class, marks a sequence point in the Microsoft intermediate language (MSIL) stream.
+        /// </summary>
+        /// <param name="document">The document for which the sequence point is being defined.</param>
+        /// <param name="startLine">The line where the sequence point begins.</param>
+        /// <param name="startColumn">The column in the line where the sequence point begins.</param>
+        /// <param name="endLine">The line where the sequence point ends.</param>
+        /// <param name="endColumn">The column in the line where the sequence point ends.</param>
```

Comment:
I think it would be helpful to indicate in a `<remarks>` section that the arguments have already been validated. Unsure if this is the common assumption for this API style.